### PR TITLE
Track2 sdk: support cert auth in multitenant credential

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azclient/auth.go [hunk 1]
```diff
@@ -128,6 +128,20 @@ func NewAuthProvider(armConfig *ARMClientConfig, config *AzureAuthConfig, client
 		if err != nil {
 			return nil, err
 		}
+		if len(armConfig.NetworkResourceTenantID) > 0 && !strings.EqualFold(armConfig.NetworkResourceTenantID, armConfig.GetTenantID()) {
+			networkClientSecretCredential, err = azidentity.NewClientCertificateCredential(armConfig.NetworkResourceTenantID, config.GetAADClientID(), certificate, privateKey, credOptions)
```
Comment (andyzhangx) [inline pkg/azclient/auth.go:132]: the logic lgtm, maybe it's better adding some logs to show it's cross tenant credential case?

Comment (MartinForReal) [reply pkg/azclient/auth.go:132]: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/internal/log/log.go is not exposed.

Comment (andyzhangx) [reply pkg/azclient/auth.go:132]: you mean we cannot log info in this func?

Comment (MartinForReal) [reply pkg/azclient/auth.go:132]: yes. the k8s logger may introduce k8s dependencies.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**